tools/rust_analyzer/aquery.rs

```diff
@@ -177,6 +177,16 @@ fn consolidate_crate_specs(crate_specs: Vec<CrateSpec>) -> anyhow::Result<BTreeS
                 existing.display_name = spec.display_name;
                 existing.crate_type = "rlib".into();
             }
+
+            // For proc-macro crates that exist within the workspace, there will be a
+            // generated crate-spec in both the fastbuild and opt-exec configuration.
+            // Prefer proc macro paths with an opt-exec component in the path.
+            if let Some(dylib_path) = spec.proc_macro_dylib_path.as_ref() {
+                const OPT_PATH_COMPONENT: &str = "-opt-exec-";
```

Comment:
Out of curiosity, is `const` doing anything here? I've only ever used it in a global scope.

Reply:
`const`s are just compiler placeholders that get inlined wherever they are used. It doesn't actually "do" anything each iteration through this loop because the &str is just inlined, but it's only used in this one function so it seemed better to keep the definition close to use than to disconnect it by putting elsewhere like the top of the file.